delfin/leader_election/tooz/leader_elector.py

```diff
@@ -0,0 +1,113 @@
+# Copyright 2021 The SODA Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Leader elector is leased based leader election"""
+
+import threading
+
+from oslo_log import log
+from oslo_utils import timeutils
+
+from delfin.coordination import LeaderElectionCoordinator
+from delfin.leader_election.interface import LeaderElector
+
+LOG = log.getLogger(__name__)
+
+
+class Elector(LeaderElector):
+
+    def __init__(self, callbacks, leader_election_key):
+        key = leader_election_key.encode('ascii')
+        super(Elector, self).__init__(callbacks, key)
+
+        self._coordinator = None
+        self.leader = False
+        self._stop = threading.Event()
+        self._runner = None
+
+    def run(self):
+        if self._coordinator:
+            return
+
+        self._stop.clear()
+
+        self._coordinator = LeaderElectionCoordinator()
+        # start leader coordinator
+        self._coordinator.start()
+
+        self._coordinator.ensure_group(self.election_key)
+        # join coordinator
+        self._coordinator.join_group()
+
+        # register callback for elected leader
+        self._coordinator. \
+            register_on_start_leading_callback(self.
+                                               callbacks.on_started_leading)
+
+        # register internal callback to notify being a leader
+        self._coordinator. \
```

Comment:
remove comments here

Reply:
This is special case for internal callback to notify current coordinator is selected as leader. 
IMO the comment is required for maintainability.